sips/sip-16/sip-016-token-metadata.md

```diff
@@ -0,0 +1,476 @@
+
+### Example
+
+token101.json
+
+```
+{
+  "sip": 16,
+  "name": "Foo #101",
+  "image": "ipfs://somerandomecid",
+  "attributes": [
+     {
+      "trait_type": "hair",
+      "value": "red",
+    },
+    {
+      "trait_type": "strength",
+      "display_type": "number",
+      "value": 99,
+    },
+  ],
+  "properties": {
+      "collection":  "Foo Collection",
+      "total_supply":  "10000"
+  },
+  "localization": {
+      "uri": "ipfs://somerandomcid/{locale}.json",
+      "default": "en",
+      "locales": ["en", "pt-BR", "de"]
+  }
+}
+```
+
+de.json
+
+```
+{
+    "sip": 16,
+    "attributes: [
+        {
+          "trait_type": "Haare",
+          "value": "rot",
+        },
+        {
+          "trait_type": "Stärke",
+          "display_type": "number",
+          "value": 99,
+        },
+    ]
+}
+```
+
+pt-BR.json
+
+```
+{
+    "sip": 16,
+    "attributes: [
+        {
+          "trait_type": "cabelos",
+          "value": "vermelho",
+        },
+        {
+          "trait_type": "amido",
+          "display_type": "number",
+          "value": 99,
+        },
+    ]
+}
+```
+
+### Properties
+
+Common properties of tokens are described in appendix C. Properties of type
+`object` are usually described using the following schema:
+
+```
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Token Metadata Property",
+    "type": "object",
+    "required": [],
+    "properties": {
+        "type": {
+            "type": "string",
+            "description": "type of the property"
+        },
+        "description": {
+            "type": "string",
+            "description": "description of the property"
+        },
+        "value": {
+            "type": {"oneOf": [{"type": "object"}, {"type": "string"}, {"type": "number"}, {"type": "integer"}, {"type": "boolean"} {"type: "array"}},
+            "description": "value of the property"
+        }
+    }
+}
+```
+
+Example:
+
+```
+{
+    "type": "string",
+    "description": "Address of custodian key holder",
+    "value": "Casa Inc., P.O. Box 20575, Charleston, S.C. 29413, United States."
+}
+```
+
+### Attributes
+
+Attributes describe additional elements of tokens that are "observable", usually
+represented in the associated image or digital asset of the token. In contrast,
+`properties` describe elements of tokens that are more abstract and not visible
+in the associated image of the token.
+
+Images of NFTs often have a limited set of traits and each trait has a limited
+number of possible values. These values are represented as attributes in the
+metadata. They can be used to calculate a score for each NFT in the collection
+that could define the rarity of the NFT.
+
+An `attribute` consists of a `trait_type` defining the name of the trait, e.g.
+"hair". The `value` is the value of the trait, e.g. "red". The `display_type` is
+a field indicating how the trait value should be displayed, e.g. on a
+marketplace.
```

Comment:
If `trait_type` and `display_type` are required fields for items in the `attributes` array, then should this be reflected in the JSON schema?

Reply:
fixed.